duckdns/data/run.sh

```diff
@@ -53,19 +53,31 @@ fi
 # Run duckdns
 while true; do
 
-    [[ ${IPV4} != *:/* ]] && ipv4=${IPV4} || ipv4=$(curl -s -m 10 "${IPV4}")
-    [[ ${IPV6} != *:/* ]] && ipv6=${IPV6} || ipv6=$(curl -s -m 10 "${IPV6}")
+    # ipv4
+    if [[ ${IPV4} == "none" ]]; then # if none, don't update ipv4 entry
+        ipv4=""
+    elif [[ ${IPV4} != *:/* ]]; then # if not a URL, use as-is
+        ipv4="&ip=${IPV4}"
+    else
+        ipv4="&ip=$(curl -s -m 10 "${IPV4}")"
+    fi
+    # ipv6
+    if [[ ${IPV6} == *:*:* ]]; then
+        ipv6="&ipv6=${IPV6}"
+    else
+        ipv6="&ipv6=$(bashio::network.ipv6_address "${IPV6}" | head -n1 | cut -d/ -f1)" || ipv6=""
+    fi
 
-    if answer="$(curl -s "https://www.duckdns.org/update?domains=${DOMAINS}&token=${TOKEN}&ip=${ipv4}&ipv6=${ipv6}&verbose=true")"; then
+    if answer="$(curl -s "https://www.duckdns.org/update?domains=${DOMAINS}&token=${TOKEN}${ipv4}${ipv6}&verbose=true")"; then
```

Comment:
I don't think it is necessary to make this change. DuckDNS used to auto-detect IPv4 if you put an empty ip like `ip=&ipv6=${ipv6}`, which was a problem if the address was CG-NAT, but that no longer seems to be the case when an ipv6 address is given.